src/env.js

```diff
@@ -73,17 +74,20 @@ export class Env {
      * Retrieves an environment variable without checking for its presence.
      * Optionally returns a default value if the environment variable doesn't
      * exist.
+     * @template [S=undefined]
      * @param {string} key The environment variable name to retrieve.
-     * @param {string} [defaultValue] The default value to return if the
+     * @param {S} [defaultValue] The default value to return if the
      *      environment variable is not found.
-     * @returns {string|undefined} The environment variable value if found or undefined if not.
+     * @returns {S extends string ? S : (string|S)} The environment variable value if found or undefined if not.
```

Comment:
Also, how would I read this type?

Reply:
This is in response to L77/79 as well.

I hope I'm explaining this OK, since I haven't really had to explain it before.

What we have here is a generic type.  _One_ way to use a generic is to change the return value of the function depending on the type of a parameter.  In this case, we always expect the `key` parameter to be a `string`, but `defaultValue` can be a `string` or `undefined` if omitted.  But given the logic of the function, the type of the return value _depends on_ the value of `defaultValue`--if it's a `string`, we always get a `string` in return.  

(This same sort of thing could be accomplished with method overloading in TS--which, of course, JS does not support, so we need to use generics.)

We define a generic in JS with a `@template` tag.  `@template X` creates a generic named `X`.  It can be of any type.  `@template {Error} Y` creates a generic `Y`, which must "extend" `Error`.  This means `Y`, in the case of `Error`, must be an `Error`, `TypeError`, `ReferenceError`, etc.  We can also provide "defaults" for generics by using the syntax `@template {Error} [Y=Error]`.  Often a generic type will _require_ type parameters; e.g., in TS, `const mySet: Set = new Set()` would be invalid, because `Set` requires a type parameter.  `const mySet: Set<string> = new Set()` would be correct (and if you ask TS to typecheck your JS, you'll get the same complaint; you need to explicitly tell TS it has a `Set<string>` by using `@type {Set<string>}`).

Here, we have `@template [S=undefined]`.  The type of the `defaultValue` param is `S` (or `undefined`, since it's optional).  The generic `S` allows us to use a ternary-lookin' _conditional type_, which is `S extends string ? S : (string|S)`.  It means **"if S looks like a string, return a value of that type (string-ish), _otherwise_ return preferably  a `string` (if the env var is set) or (if the env var is unset) whatever `S` is (typically `undefined`)**.

The outcome is that TS's inference can easily tell if you call `Env.get` without a second parameter, and it deduces that the return type must be `string|undefined`.  But if the second parameter is a `string`, then we must get a `string` return type.

Reply:
Ah, thank you. Makes sense now. 👍

Reply:
I think this change could also apply to the `first()` method, but I'm not using that one, so I didn't change it. If you want me to, I can do that as well.

Reply:
Yeah, please go ahead with that change. 👍

Reply:
This is done